### PR TITLE
[Bug] Hunting - Add UTF-8 Encoding for all Read and Write Operations

### PR DESCRIPTION
**Pull request by terrancedejesus**
# Pull Request

*Issue link(s)*:
* https://github.com/elastic/ia-trade-team/issues/280

## Summary - What I changed

⚡ A list of hunting queries (see below) had `ñ` characters in the regex when normalizing paths in ES|QL queries. This was added intentionally to match on that character. However, when running `hunting/generate_markdown.py` script, we did not include UTF-8 encoding on the content when loading the TOML file, only when writing to Markdown. As a result @Aegrah [pointed out](https://github.com/elastic/detection-rules/pull/3872#pullrequestreview-2165516644) that the `ñ` changed when writing to the Markdown files from a Windows host.

This is likely because UTF-8 was used on the writing operations and not the loading.

List of hunts affected:
- Low Occurrence of Suspicious Launch Agent or Launch Daemon
- Low Occurrence Rate of CreateRemoteThread by Source Process
- Low Occurrence of Drivers Loaded on Unique Hosts
- Persistence via Run Key with Low Occurrence Frequency
- Unique Windows Services Creation by Service File Name

Note that the hunts listed, both TOML and MD files, are correct in main and do not require further adjustment.

## How To Test

1. From a Windows host, checkout the main
2. Run `python ~/detection-rules/hunting/generate_markdown.py`
3. Notice the diff on the queries listed above
4. Remove changes
5. Checkout `bug-hunting-encoding-content-from-files` branch
6. Run  `python ~/detection-rules/hunting/generate_markdown.py`
7. Notice no changes

hunt-utf

## Checklist

- [x] Added a label for the type of pr: `bug`, `enhancement`, `Rule: New`, `Rule: Deprecation`, `Rule: Promote`, `Rule: Tuning`, `Hunt: New`, or `Hunt: Tuning` so guidelines can be generated
- [ ] Added the `meta:rapid-merge` label if planning to merge within 24 hours
- [x] Secret and sensitive material has been managed correctly
- [x] Automated testing was updated or added to match the most common scenarios
- [x] Documentation and comments were added for features that require explanation

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

**Comment by protectionsmachine** [comment]
## Bug - Guidelines

These guidelines serve as a reminder set of considerations when addressing a bug in the code.

### Documentation and Context

- [ ] Provide detailed documentation (description, screenshots, reproducing the bug, etc.) of the bug if not already documented in an issue.
- [ ] Include additional context or details about the problem.
- [ ] Ensure the fix includes necessary updates to the release documentation and versioning.

### Code Standards and Practices

- [ ] Code follows established design patterns within the repo and avoids duplication.
- [ ] Code changes do not introduce new warnings or errors.
- [ ] Variables and functions are well-named and descriptive.
- [ ] Any unnecessary / commented-out code is removed.
- [ ] Ensure that the code is modular and reusable where applicable.
- [ ] Check for proper exception handling and messaging.

### Testing

- [ ] New unit tests have been added to cover the bug fix or edge cases.
- [ ] Existing unit tests have been updated to reflect the changes.
- [ ] Provide evidence of testing and detecting the bug fix (e.g., test logs, screenshots).
- [ ] Validate that any rules affected by the bug are correctly updated.
- [ ] Ensure that performance is not negatively impacted by the changes.
- [ ] Verify that any release artifacts are properly generated and tested.

### Additional Checks

- [ ] Ensure that the bug fix does not break existing functionality.
- [ ] Review the bug fix with a peer or team member for additional insights.
- [ ] Verify that the bug fix works across all relevant environments (e.g., different OS versions).
- [ ] Confirm that all dependencies are up-to-date and compatible with the changes.

File: hunting/generate_markdown.py [hunk 1]
```diff
@@ -50,7 +50,9 @@ def load_all_toml(base_path: Path) -> List[tuple[Hunt, Path]]:
     """Load all TOML files from the directory and return a list of Hunt configurations and their paths."""
     hunts = []
     for toml_file in base_path.rglob("*.toml"):
-        hunt_config = load_toml(toml_file.read_text())
+        with toml_file.open('r', encoding='utf-8') as file:
```
Comment (shashank-elastic) [inline hunting/generate_markdown.py:53]:
```suggestion
        hunt_config = load_toml(toml_file.read_text(encoding='utf-8'))
```

Comment (shashank-elastic) [reply hunting/generate_markdown.py:53]: Did we try this? Just curious why we reverted to open file and read mode when we could pass encoding to the read_text option.

**Review by shashank-elastic — APPROVED** [review]

File: hunting/generate_markdown.py (hunk 1 above)
Comment (terrancedejesus) [reply hunting/generate_markdown.py]: Great observation, reverted and made the minor tweak!

**Review by Mikaayenson — APPROVED** [review]
